src/Umbraco.Web.UI.Client/src/common/directives/components/content/edit.controller.js

```diff
@@ -468,15 +469,19 @@
 
                 eventsService.emit("content.saved", { content: $scope.content, action: args.action });
 
-                resetNestedFieldValiation(fieldsToRollback);
+                if($scope.contentForm.$invalid !== true) {
+                    resetNestedFieldValiation(fieldsToRollback);
+                }
                 ensureDirtyIsSetIfAnyVariantIsDirty();
 
                 return $q.when(data);
             },
                 function (err) {
                     syncTreeNode($scope.content, $scope.content.path);
 
-                    resetNestedFieldValiation(fieldsToRollback);
+                    if($scope.contentForm.$invalid !== true) {
```

Comment:
Ah so much inversion - not an easy line to read, so basically the form is valid then?! 🤯

Reply:
Yep, if it is valid then we performe this if.